### DISCUSSION TITLE
Issue with when scrolling a ListBox/VirtualizingStackPanel when hidden

### DISCUSSION THREAD

**Discussion by BobLd** [AvaloniaUI/Avalonia#15188] [Q&A]
_I originally posted this question on the Telegram channel and was advised to create a Q&A._

It seems there's an issue with `ListBox` (due to `VirtualizingStackPanel` I think) when scrolling the list while it is hidden. I've created a minimum app that reproduces the issue, available here https://github.com/BobLd/VirtualizingStackPanelDuplicatesIssue

To replicate the issue:
1. Click the `Scroll down` button to scroll down the listbox while it is not visible
2. Click `Toggle Menubar` to show the list box
3. Get the messy result:

![image](https://github.com/AvaloniaUI/Avalonia/assets/38405645/68a87524-73f2-459c-b152-315ea41741cb)

I originally encountered the problem in https://github.com/BobLd/Caly/tree/develop-menubar

**Comment by doublecount** [comment]
I have that same experience in other situations than scrolling.
I do get around with having the ItemsPresenter itself centered horizontally inside its parent (ScrollViewer). In this case the sometimes doubled visuals are on top of each other. But of course it would be nicer to have no duplicated visuals.

**Comment by timunie** [comment]
1. The sample is a great reproduction 👍 
2. Sadly, seems to be a bug indeed. So will convert this into and issue report. I don't know when this can be fixed as the root cause need to be found.

In case anyone want's to debug it, I suggest to add the sample into src/Samples/Sandbox and check also #13736 for other similar issues and fixes.
